### PR TITLE
Rewrite Dashboard query with kysely provider

### LINKED ISSUE #2653: Rewrite Dashboard query with `kysely` provider

### PR DESCRIPTION
**Pull request by LamaEats**
Use kysely query for dashboard instead of prisma

## PR includes

- [x] Refactor
- [x] DX

## Related issues

Resolve #2653

### CODE REVIEW

File: src/components/DashboardPage/DashboardPage.tsx [hunk 1]
```diff
@@ -35,29 +33,26 @@ export const DashboardPage = ({ user, ssrTime, defaultPresetFallback }: External
...
-        trpc.project.getUserProjectsWithGoals.useInfiniteQuery(
-            {
-                limit: projectsLimit,
-                goalsQuery: queryState,
-            },
+        trpc.v2.project.userProjectsWithGoals.useInfiniteQuery(
+            { goalsQuery: queryState },
             {
-                getNextPageParam: (p) => p.nextCursor,
+                getNextPageParam: (p) => p.pagiantion.offset,
```
Comment (tsumo) [inline src/components/DashboardPage/DashboardPage.tsx:39]: typo, pagination

**Review by tsumo — APPROVED** [review]
I can't really read all those queries, I guess we all are gonna need to learn some SQL soon...

**Review by asabotovich — APPROVED** [review]

File: trpc/queries/goalV2.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+                                    selectFrom('_projectAccess')
+                                        .select('B')
+                                        .where('_projectAccess.A', '=', params.activityId),
+                                ),
+                            ),
+                        ]),
+                    ),
+            ),
+        )
+        .limit(30)
```
Comment (9teen90nine) [inline trpc/queries/goalV2.ts:207]: you probably should move it higher to a const or even an ENV

**Review by 9teen90nine — APPROVED** [review]

File: trpc/queries/goalV2.ts (hunk 2 above)
Comment (LamaEats) [reply trpc/queries/goalV2.ts:207]: This code now is useless, in projects' query module I created same variable

**Pull request merged**